src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -660,6 +669,49 @@ private EEMethodSymbol GetTypeVariablesMethod(EENamedTypeSymbol container, strin
             return new BoundReturnStatement(syntax, RefKind.None, expression) { WasCompilerGenerated = true };
         }
 
+        private static bool TryBindFieldKeyword(Symbol? containingMemberOrLambda, ExpressionSyntax syntax, [NotNullWhen(true)] out BoundFieldAccess? boundField)
```

Comment:
>TryBindFieldKeyword [](http://example.com/codeflow?start=28&length=19)

This doesn't feel like the right way to address the problem:
- It only handles standalone identifier.
- The identifier might actually refer to a local in scope even when there is a backing field.
- Simply relying on name feels fragile. 

Perhaps the field in metadata should be specially marked (with an attribute?) and the binder chain should be adjusted to bring it in scope at the right time. However, as I already suggested earlier, let's not try to fix/change behavior just yet. Let's focus on adding tests and identifying questionable/unexpected behavior.